eth-contracts/contracts/service/DelegateManager.sol

```diff
@@ -0,0 +1,556 @@
+pragma solidity ^0.5.0;
+import "../staking/Staking.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20Mintable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "./registry/RegistryContract.sol";
+import "./interface/registry/RegistryInterface.sol";
+
+import "../staking/Staking.sol";
+import "./ServiceProviderFactory.sol";
+import "./ClaimFactory.sol";
+
+
+// WORKING CONTRACT
+// Designed to manage delegation to staking contract
+contract DelegateManager is RegistryContract {
+    using SafeMath for uint256;
+    RegistryInterface registry = RegistryInterface(0);
+
+    address tokenAddress;
+    address stakingAddress;
+
+    bytes32 stakingProxyOwnerKey;
+    bytes32 serviceProviderFactoryKey;
+    bytes32 claimFactoryKey;
+
+    // Number of blocks an undelegate operation has to wait
+    // TODO: Expose CRUD
+    // TODO: Move this value to Staking.sol as SPFactory may need as well
+    uint undelegateLockupDuration = 10;
```

Comment:
v1 - Expose CRUD and make this governable. Consider movement to Staking.sol separately